diesel_derives/src/sql_function.rs

```diff
@@ -123,7 +125,39 @@ pub(crate) fn expand(input: SqlFunctionDecl) -> Result<TokenStream, Diagnostic>
         }
     };
 
-    if !is_aggregate {
+    if is_aggregate {
+        if cfg!(feature = "sqlite") {
+            let type_params = &generics
+                .type_params()
+                .filter(|type_param| !type_param.bounds.is_empty())
+                .cloned()
+                .collect::<Vec<_>>();
+
+            tokens = quote! {
+                #tokens
+
+                use diesel::sqlite::{Sqlite, SqliteConnection};
+                use diesel::serialize::ToSql;
+                use diesel::deserialize::Queryable;
+                use diesel::sqlite::Aggregator;
+
+                #[allow(dead_code)]
+                pub fn register_impl<A, #(#arg_name),*, #(#type_args),*>(
+                    conn: &SqliteConnection
+                ) -> QueryResult<()>
+                where
+                     A: Aggregator<#(#arg_name),*, Output=ToSql<#return_type, Sqlite>> + Send + 'static,
+                     (#(#arg_name,)*): Queryable<(#(#arg_type,)*), Sqlite>,
+                     #(#type_params),*
```

Comment:
I've been having some difficulty with this. For example, in `diesel/src/expression/functions/aggregate_folding.rs`:
```
sql_function! {
    #[aggregate]
    fn sum<ST: Foldable>(expr: ST) -> ST::Sum;
}
```

What I've written does indeed generate `ST: Foldable` in the where clause. However, I'm receiving compile errors like this:
```
error[E0277]: the size for values of type `(dyn serialize::ToSql<<ST as sql_types::fold::Foldable>::Sum, sqlite::backend::Sqlite> + 'static)` cannot be known at compilation time
   --> diesel/src/expression/functions/mod.rs:167:9
    |
167 |           sql_function_proc! { $($args)* }
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
    |
   ::: diesel/src/expression/functions/aggregate_folding.rs:3:1
    |
3   | / sql_function! {
4   | |     /// Represents a SQL `SUM` function. This function can only take types which are
5   | |     /// Foldable.
6   | |     ///
...   |
21  | |     fn sum<ST: Foldable>(expr: ST) -> ST::Sum;
22  | | }
    | |_- in this macro invocation
    |
    = help: the trait `std::marker::Sized` is not implemented for `(dyn serialize::ToSql<<ST as sql_types::fold::Foldable>::Sum, sqlite::backend::Sqlite> + 'static)`
    = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
note: required by `sqlite::connection::raw::Aggregator`
   --> diesel/src/sqlite/connection/raw.rs:14:1
    |
14  | pub trait Aggregator<Args>: Default {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0038]: the trait `serialize::ToSql` cannot be made into an object
   --> diesel/src/expression/functions/mod.rs:167:9
    |
167 |           sql_function_proc! { $($args)* }
    |           ^^^^^^^^^^^^^^^^^^ the trait `serialize::ToSql` cannot be made into an object
    |
   ::: diesel/src/expression/functions/aggregate_folding.rs:3:1
    |
3   | / sql_function! {
4   | |     /// Represents a SQL `SUM` function. This function can only take types which are
5   | |     /// Foldable.
6   | |     ///
...   |
21  | |     fn sum<ST: Foldable>(expr: ST) -> ST::Sum;
22  | | }
    | |_- in this macro invocation
    |
    = note: method `to_sql` has generic type parameters
```

Do you have any advice for this?